crates/pallets/starknet/src/execution_config.rs

```diff
@@ -8,47 +9,40 @@ use crate::Config;
 /// flags for each transaction mode (e.g. normal, estimate fee, simulate, ...).
 pub struct RuntimeExecutionConfigBuilder(ExecutionConfig);
 
-impl Default for RuntimeExecutionConfigBuilder {
-    fn default() -> Self {
+impl RuntimeExecutionConfigBuilder {
+    pub fn new<T: Config>() -> Self {
         Self(ExecutionConfig {
             is_query: false,
-            disable_transaction_fee: false,
             disable_fee_charge: false,
-            disable_nonce_validation: false,
             disable_validation: false,
+            disable_nonce_validation: T::DisableNonceValidation::get(),
+            disable_transaction_fee: T::DisableTransactionFee::get(),
         })
     }
-}
-
-impl RuntimeExecutionConfigBuilder {
+    #[must_use]
     pub fn with_query_mode(mut self) -> Self {
         self.0.is_query = true;
         self
     }
-    pub fn with_transaction_fee_disabled(mut self) -> Self {
-        self.0.disable_transaction_fee = true;
-        self
-    }
-    pub fn with_fee_charge_disabled(mut self) -> Self {
-        self.0.disable_fee_charge = true;
-        self
-    }
-    pub fn with_nonce_validation_disabled(mut self) -> Self {
-        self.0.disable_nonce_validation = true;
-        self
-    }
-    pub fn with_validation_disabled(mut self) -> Self {
-        self.0.disable_validation = true;
+    #[must_use]
+    pub fn with_simulation_mode(mut self, simulation_flags: &[SimulationFlag]) -> Self {
```

Comment:
having them as a vector when there are never more than two is not great. 
What is the reason for this rather than a Struct with two fields?

If there is no good reason, can you open an issue? It will be a good simple first issue